### ISSUE TITLE
Improve installation and contribution

### ISSUE THREAD

**Issue by melsophos** [ribault/bootstrap-2d-Python#3]
In order to improve the installation and contributions from other users it could be useful to provide the following tools:

1. Generate the `.py` modules from the `.ipynb` notebooks (avoids `ModuleNotFoundError`).
2. Clean the cell ouputs in order to be git-friendly (see for example [this site](http://timstaley.co.uk/posts/making-git-and-jupyter-notebooks-play-nice/)).
3. Have a `.gitignore` file with `*.py` to reconcile point 2) and 3).
4. Modify the `README` file to explain how to use these tools.

Points 1) and 2) can be achieved thanks to a script.

**Comment by ribault** [comment]
Thank you for the issue and for the associated commit. Your script 'manage.sh' may well be useful to some people. 
I am not sure how useful the .gitignore file is in its present state: if you wanted to do something like 'git add *' you would have to ignore more than just .py files. But it does not hurt to have that file, and it can be later expanded.  
I am not considering closing this issue for the moment, as this is a good place for discussing further improvements along these lines.

**Comment by melsophos** [comment]
> I am not sure how useful the .gitignore file is in its present state: if you wanted to do something like 'git add *' you would have to ignore more than just .py files.

Indeed, I did not take this into account because my git configuration already ignores all such additional files (which any good git initialisation should do), but for safety, one may wish to include everything.
To avoid problems I generically don't use `git add *` but manually select the new files to be added (which is viable for such a small project). Then I update with `git ci -a` such that only tracked files are updated. The gitignore is mostly useful for having a clear list of what changed in my opinion.
